### PR TITLE
rpc: allow specifying min chain depth for inputs in fund calls

### PR DESCRIPTION
**Pull request by champo**
Enables users to craft BIP-125 replacements with changes to the output
list, ensuring that if additional funds are inputs they will be added.

### CODE REVIEW

File: test/functional/rpc_fundrawtransaction.py [hunk 1]
```diff
@@ -969,6 +970,53 @@ def test_include_unsafe(self):
...
 
+    def test_inputs_min_chain_depth(self):
+        unconfirmedAddress = self.nodes[0].getnewaddress()
+        targetAddress = self.nodes[2].getnewaddress()
+        unconfirmedTxid = self.nodes[0].sendtoaddress(unconfirmedAddress, 1)
+
+        utxo1 = self.nodes[0].listunspent(0, 1, [unconfirmedAddress])[0]
+        assert unconfirmedTxid == utxo1['txid']
+
+        self.log.info("Crafting PSBT using an unconfirmed input")
```
Comment (achow101) [inline test/functional/rpc_fundrawtransaction.py:981]: In 6086c58d27146d500b7ff8de89a5279a6f88457d "rpc: allow specifying min chain depth for inputs in fund calls"

```suggestion
        self.log.info("Crafting tx using an unconfirmed input")
```

Here and elsewhere in this file too.

File: src/wallet/rpcwallet.cpp [hunk 2]
```diff
@@ -3215,6 +3224,7 @@ static RPCHelpMan fundrawtransaction()
                             {"include_unsafe", RPCArg::Type::BOOL, RPCArg::Default{false}, "Include inputs that are not safe to spend (unconfirmed transactions from outside keys and unconfirmed replacement transactions).\n"
                                                           "Warning: the resulting transaction may become invalid if one of the unsafe inputs disappears.\n"
                                                           "If that happens, you will need to fund the transaction with different inputs and republish it."},
+                            {"inputs_min_depth", RPCArg::Type::NUM, RPCArg::Default{0}, "If add_inputs is specified, require inputs with at least this chain depth."},
```
Comment (promag) [inline src/wallet/rpcwallet.cpp:3227]: IMO should be `minconf`, see `listunspent` RPC.

**Review by achow101 — COMMENTED** [review]
Seems reasonable, although the use case for this is not immediately obvious to me.

Perhaps it would also be useful to add `inputs_max_depth` too?

**Review by promag — COMMENTED** [review]
Concept ACK.

**Comment by champo** [comment]
> Seems reasonable, although the use case for this is not immediately obvious to me.

The use case is for applications that are using RBF for something else than bumping fee (as implemented by `bumpfee` RPC). RBF requires new inputs added to a replacement to be confirmed. The functional tests are modeled after one of the use cases: increasing the value of an output beyond the current funding of the TX.

> Perhaps it would also be useful to add `inputs_max_depth` too?

Will do!

**Comment by champo** [comment]
Thanks for the quick look guys! I've updated the changset with your feedback. I'm not sure what the ettiquete is here regarding squashing mid-review, so I opted to add new commits. LMK if this is not what you expect.

File: src/wallet/rpcwallet.cpp [hunk 3]
```diff
@@ -4043,6 +4063,8 @@ static RPCHelpMan send()
                     {"include_unsafe", RPCArg::Type::BOOL, RPCArg::Default{false}, "Include inputs that are not safe to spend (unconfirmed transactions from outside keys and unconfirmed replacement transactions).\n"
                                                           "Warning: the resulting transaction may become invalid if one of the unsafe inputs disappears.\n"
                                                           "If that happens, you will need to fund the transaction with different inputs and republish it."},
+                    {"minconfs", RPCArg::Type::NUM, RPCArg::Default{0}, "If add_inputs is specified, require inputs with at least this confirmations."},
```
Comment (promag) [inline src/wallet/rpcwallet.cpp:4066]: It's singular:
https://github.com/bitcoin/bitcoin/blob/b295395664bd37e26d168c329f238237b34aef8c/src/wallet/rpcwallet.cpp#L2844-L2845

**Comment by luke-jr** [comment]
Duplicate of #14641 ?

**Comment by luke-jr** [comment]
Rebased & squashed here: https://github.com/bitcoin/bitcoin/compare/master...luke-jr:rpc_fundtx_minmaxconf

**Comment by fanquake** [comment]
@champo would you like to follow up here?

**Comment by glozow** [comment]
Closing as this has needed rebase for more than 1 year. Feel free to reopen if you get a chance to work on this again in the future, thanks!

**Pull request closed without merging**